tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/TrainingNumberServiceImpl.java

```diff
@@ -117,4 +136,293 @@ public void delete(Long id) {
+
+    if (isExcluded(programmeMembership)) {
+      return;
+    }
+
+    String parentOrganization = getParentOrganization(programmeMembership);
+    String specialtyConcat = getSpecialtyConcat(programmeMembership);
+    String referenceNumber = getReferenceNumber(programmeMembership.getPerson());
+    String suffix = getSuffix(programmeMembership);
+
+    TrainingNumber trainingNumber = new TrainingNumber();
+    trainingNumber.setTrainingNumber(
+        parentOrganization + "/" + specialtyConcat + "/" + referenceNumber + "/" + suffix);
+    programmeMembership.setTrainingNumber(trainingNumber);
+    log.info("Populated training number: {}.", trainingNumber);
+  }
+
+
+  /**
+   * Get the parent organization for the given programme membership.
+   *
+   * @param programmeMembership The programme membership to calculate the parent organization for.
+   * @return The calculated parent organization.
+   */
+  private String getParentOrganization(ProgrammeMembership programmeMembership) {
+    String managingDeanery = programmeMembership.getProgramme().getOwner();
+    log.info("Calculating parent organization for managing deanery '{}'.", managingDeanery);
+    String parentOrganization = null;
+
+    if (managingDeanery != null) {
+      switch (managingDeanery) {
+        case "Defence Postgraduate Medical Deanery":
+          parentOrganization = "TSD";
+          break;
+        case "Health Education England East Midlands":
+          parentOrganization = "EMD";
+          break;
+        case "Health Education England East of England":
+          parentOrganization = "EAN";
+          break;
+        case "Health Education England Kent, Surrey and Sussex":
+          parentOrganization = "KSS";
+          break;
+        case "Health Education England North Central and East London":
+        case "Health Education England South London":
+        case "Health Education England North West London":
+        case "London LETBs":
+          parentOrganization = "LDN";
+          break;
+        case "Health Education England North East":
+          parentOrganization = "NTH";
+          break;
+        case "Health Education England North West":
+          parentOrganization = "NWE";
+          break;
+        case "Health Education England South West":
+          parentOrganization = "SWN";
+          break;
+        case "Health Education England Thames Valley":
+          parentOrganization = "OXF";
+          break;
+        case "Health Education England Wessex":
+          parentOrganization = "WES";
+          break;
+        case "Health Education England West Midlands":
+          parentOrganization = "WMD";
+          break;
+        case "Health Education England Yorkshire and the Humber":
+          parentOrganization = "YHD";
+          break;
+        default:
+          break;
+      }
+    }
+
+    if (parentOrganization == null) {
+      throw new IllegalArgumentException("Unable to calculate the parent organization.");
+    }
```

Comment:
So it keeps it to 1 throw statement rather than throwing "no programme owner", setting the parent org or default to throwing "invalid programme owner".  OK.